tracing-mock/src/expect.rs

```diff
@@ -19,3 +19,25 @@ pub(crate) enum Expect {
     NewSpan(NewSpan),
     Nothing,
 }
+
+pub fn event() -> ExpectedEvent {
+    ExpectedEvent {
+        ..Default::default()
+    }
+}
+
+pub fn field<K>(name: K) -> ExpectedField
+where
+    String: From<K>,
+{
+    ExpectedField {
+        name: name.into(),
+        value: ExpectedValue::Any,
+    }
+}
+
+pub fn span() -> ExpectedSpan {
+    ExpectedSpan {
+        ..Default::default()
+    }
+}
```

Comment:
this is unrelated to this PR, but it occurs to me that these probably ought to have `#[must_use]` annotations on them, since calling these functions and dropping the result will do nothing. we should probably do a pass to add `#[must_use]` annotations as part of the release prep, in a follow-up PR...